packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -306,7 +306,7 @@ module.exports = {
                         minimize: true,
                         sourceMap: shouldUseSourceMap,
                         modules: true,
-                        localIdentName: '[path]__[name]___[local]',
```

Comment:
As per suggestions from Dan, you can try something along these lines for both dev and prod.

```js
{
  loader: require.resolve('css-loader'),
  options: {
    importLoaders: 1,
    modules: true,
    getLocalIdent: (context, localIdentName, localName, options) => {
      // Use the filename or folder name, based on some uses the index.js / index.module.css project style
      const fileNameOrFolder = context.resourcePath.endsWith('index.module.css') ? '[folder]' : '[name]'
      // Create a hash based on a the file location and class name. Will be unique across a project, and close to globally unique.
      const hash = loaderUtils.getHashDigest(context.resourcePath + localName, 'md5', 'base64', 5)
      // Use loaderUtils to find the file or folder name
      const className = loaderUtils.interpolateName(context, fileNameOrFolder + '_' + localName + '__' + hash , options)
      // remove the .module that appears in every classname when based on the file.
      return className.replace('.module_', '_')
    },
    minimize: true,
    sourceMap: shouldUseSourceMap,
  },
},
```

Reply:
We should also "hide" this function into `react-dev-utils/getCSSModuleLocalIdent` so that it doesn't get duplicated in both places.

Reply:
@ro-savage Thank you so much. Was on my todo list to use `getLocalIdent` to solve the  `src/component/button/index.module.css` or `src/component/button/button.module.css` class naming convention headache.